src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.intakeVoltage.get()
+          )
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+            Manipulator.TunableManipulatorStates.cubeInVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP -> {
+        // Goes immediately to IDLE
+        // Outputs
+
+        // Transition
+        nextState = SuperstructureStates.IDLE
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedUpAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+            Manipulator.TunableManipulatorStates.coneInVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP -> {
```

Comment:
reminder that you may want some weird stuff here with moving the elevator up before you retract so that the cone doesn't clip stuff -- depending on IRL testing